source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/GenerateFormContainer.tsx

```diff
@@ -14,16 +22,143 @@ export interface IGenerateFormContainerProps {
+        if (fileOwners !== undefined) {
+          fileOwners?.map(owner =>
+            generateRecipientsList.push(
+              new LetterRecipientModel(owner.id!, 'OWNR', new Api_GenerateOwner(owner), owner),
+            ),
+          );
+        }
+
+        if (intHolders !== undefined) {
+          intHolders.map(async holder => {
+            if (holder.personId) {
+              const person = (await getPersonConcept(holder?.personId))?.data;
+              if (person) {
+                generateRecipientsList.push(
+                  new LetterRecipientModel(
+                    holder.interestHolderId!,
+                    'HLDR',
+                    Api_GenerateOwner.fromApiPerson(person),
+                    holder,
+                  ),
+                );
+              }
+            } else if (holder.organizationId) {
+              const org = (await getOrganization(holder?.organizationId))?.data;
+              if (org) {
+                generateRecipientsList.push(
+                  new LetterRecipientModel(
+                    holder.interestHolderId!,
+                    'HLDR',
+                    Api_GenerateOwner.fromApiOrganization(org),
+                    holder,
+                  ),
+                );
+              }
+            }
+          });
+        }
+      },
+    );
+
+    // Add owners solicitors to recipients list
+    if (file.acquisitionFileOwnerSolicitors && file.acquisitionFileOwnerSolicitors.length > 0) {
+      await Promise.all(
+        file.acquisitionFileOwnerSolicitors.map(async solicitor => {
+          if (solicitor.personId) {
+            const person = (await getPersonConcept(solicitor?.personId))?.data;
+            if (person) {
+              const personSolicitorModel = Api_GenerateOwner.fromApiPerson(person);
+              generateRecipientsList.push(
+                new LetterRecipientModel(person.id!, 'SLTR', personSolicitorModel, solicitor),
+              );
+            }
+          } else if (solicitor.organizationId) {
+            const org = (await getOrganization(solicitor?.organizationId))?.data;
+            if (org) {
+              const orgSolicitorModel = Api_GenerateOwner.fromApiOrganization(org);
+              generateRecipientsList.push(
+                new LetterRecipientModel(org.id!, 'SLTR', orgSolicitorModel, solicitor),
+              );
+            }
+          }
+        }),
+      );
+    }
+
+    //Add Owner Representatives
+    if (
+      file.acquisitionFileOwnerRepresentatives &&
+      file.acquisitionFileOwnerRepresentatives.length > 0
+    ) {
+      await Promise.all(
+        file.acquisitionFileOwnerRepresentatives.map(async rep => {
+          if (rep.personId) {
+            const person = (await getPersonConcept(rep?.personId))?.data;
+            if (person) {
+              const personSolicitorModel = Api_GenerateOwner.fromApiPerson(person);
+              generateRecipientsList.push(
```

Comment:
My main comment here is why are we using GenerateOwner instances here for owner solicitors and owner representatives. We do have GeneratePerson - why not use that instead??

Reply:
The array passed to the template are not 'Person/Contacts' but GenerateOwners.

Reply:
I see. We could leave as-is but there's nothing stopping us from altering the template to take a list of owners, and a list of additional "contacts" and then display them one after another...
Like in Expropriation forms we have many different "groups" of people that receive the letter and they are not all made into owners to fit one big list. See below. 

![image](https://github.com/bcgov/PSP/assets/24322224/64de477c-63dc-48d5-bf4b-f5c21ef9c0cc)

Not saying you must do it this way but it would be cleaner to pass in the extra people as contacts than "force" then into GenerateOwner instances when they don't fit naturally, but again a matter of dev preference

Reply:
It turned out to be much simpler with the new interest holder logic. now just mapping an interest holder to a generateOwner ( instead of mapping an Owner to a Contact ). didn't wanted to modify the template because at pointing we decided not to modify the template and I rather keep the template simple since it's managed by users.